### ISSUE TITLE
`test_expectedFDs` doesn't need to look at stdin, stdout, stderr

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5012]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5012|
|Type|defect|
|Created|2011-04-03 01:10:47Z|
|Branch|https://github.com/twisted/twisted/tree/expectedfds-test-5012|

This test uses stdin, stdout, and stderr to find three file descriptors that it thinks should be open.  Certainly these descriptors are quite often open, but there is no guarantee that they will be.  For example, disttrial ([#1784](https://github.com/twisted/twisted/issues/1784)) runs tests in a child process and talks to it over an AMP connection.  stdio isn't hooked up to anything meaningful.

<details><summary>Searchable metadata</summary>

```
trac-id__5012 5012
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_expectedfds_test_5012 branches/expectedfds-test-5012
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1301793047000000 1301793047000000
changetime__1301854656000000 1301854656000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [31522]) Branching to 'expectedfds-test-5012'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Made some changes, not too complicated, [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/expectedfds-test-5012) look good.

**Comment by twisted-trac** [comment]
|[<img alt="mithrandi's avatar" src="https://avatars.githubusercontent.com/u/856430?s=50" width="50" height="50">](https://github.com/mithrandi)<a name="note_3"></a>|@mithrandi set owner to @exarkun|
|-|-|

There's a pyflakes warning introduced by this change:
```
twisted/internet/test/test_posixprocess.py:8: 'sys' imported but unused
```

Please go ahead and merge once this is fixed.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [31540]) Merge expectedfds-test-5012

Author: exarkun
Reviewer: mithrandi
Fixes: [#5012](https://github.com/twisted/twisted/issues/5012)

Change `twisted.internet.test.test_posixprocess.FileDescriptorTests.test_expectedFDs`
to not rely on `sys.stdin`, `sys.stdout`, and `sys.stderr`.  Instead, just manipulate
a newly opened file descriptor.

This helps reduce the dependence of this test on its execution environment, and in
particular lets it work if stdio is not connected to anything.
